specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/ServiceFabricManagedClusters/ApplicationResource.tsp

```diff
@@ -618,3 +666,108 @@ model RuntimeUpdateApplicationUpgradeParameters {
+   */
+  Ok: "Ok",
+
+  /**
+   * Filter for health state Warning.
+   */
+  Warning: "Warning",
+
+  /**
+   * Filter for health state Error.
+   */
+  Error: "Error",
+
+  /**
+   * Filter that matches input with any health state.
+   */
+  All: "All",
+}
+
+/**
+ * Parameters for fetching the health of an application.
+ */
+@added(Versions.v2025_10_01_preview)
+model ApplicationFetchHealthRequest {
+  /**
+   * Allows filtering of the health events returned in the response based on health state.
+   */
+  eventsHealthStateFilter?: HealthFilter;
+
+  /**
+   * Allows filtering of the deployed applications health state objects returned in the result of application health query based on their health state.
+   */
+  deployedApplicationsHealthStateFilter?: HealthFilter;
+
+  /**
+   * Allows filtering of the services health state objects returned in the result of services health query based on their health state.
+   */
+  servicesHealthStateFilter?: HealthFilter;
+
+  /**
+   * Indicates whether the health statistics should be returned as part of the query result. False by default. The statistics show the number of children entities in health state Ok, Warning, and Error.
+   */
+  excludeHealthStatistics?: boolean = false;
+
+  /**
+   * Request timeout for the health query in seconds. The default value is 60 seconds.
+   */
+  timeout?: int64 = 60;
+}
+
+/**
+ * Parameters for restarting a deployed code package.
+ */
+@added(Versions.v2025_10_01_preview)
+model RestartDeployedCodePackageRequest {
+  /**
+   * The name of the node where the code package needs to be restarted. Use '*' to restart on all nodes where the code package is running.
+   */
+  nodeName: string;
+
+  /**
+   * The name of the service manifest as specified in the code package.
+   */
+  serviceManifestName: string;
+
+  /**
+   * The name of the code package as specified in the service manifest.
+   */
+  codePackageName: string;
+
+  /**
+   * The instance ID for currently running entry point. For a code package setup entry point (if specified) runs first and after it finishes main entry point is started. Each time entry point executable is run, its instance ID will change. If 0 is passed in as the code package instance ID, the API will restart the code package with whatever instance ID it is currently running. If an instance ID other than 0 is passed in, the API will restart the code package only if the current Instance ID matches the passed in instance ID. Note, passing in the exact instance ID (not 0) in the API is safer, because if ensures at most one restart of the code package.
+   */
+  codePackageInstanceId: string;
+
+  /**
+   * The activation id of a deployed service package. If ServicePackageActivationMode specified at the time of creating the service is 'SharedProcess' (or if it is not specified, in which case it defaults to 'SharedProcess'), then value of ServicePackageActivationId is always an empty string.
+   */
+  servicePackageActivationId?: string;
```

Comment:
>servicePackageActivationId [](http://example.com/codeflow?start=2&length=26)

Similar issue here: can you clarify for clients what kinds of values this property takes (with type enforcement as possible/appropriate)?